### PR TITLE
Fix duplicated labels in column picker

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it fixes:
- duplicated labels in column picker
- duplicated queries execution in different dashboard places by migrating `react-infinite-scroller` to `react-infinite-scroll-component`

**PR intended to be tested with API branch:**  master

### Screenshots

Zrzut ekranu 2021-06-28 o 18 07 40
vs.
Zrzut ekranu 2021-06-28 o 18 07 32

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [x] Attributes `[data-test-id]` are added for new elements
4. [x] Changes are mentioned in the changelog
5. [x] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://qa.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-3618 Duplicated labels](https://app.clickup.com/t/2549495/SALEOR-3618)

**Review by eCzerniak — APPROVED** [review]

File: src/components/ColumnPicker/ColumnPickerContent.tsx [hunk 1]
```diff
@@ -142,7 +148,7 @@ const ColumnPickerContent: React.FC<ColumnPickerContentProps> = props => {
                   key={column.value}
                 />
               ))}
-              {loading && (
+              {hasMore && loading && (
```
Comment (jwm0) [inline src/components/ColumnPicker/ColumnPickerContent.tsx:151]: Shouldn't we use the built in functionality for that?
```
<CardContent>
  <div id={scrollableTargetId}>
    <InfiniteScroll
      loader={<div className={classes.loadMoreLoaderContainer}><CircularProgress size={16} /></div>}
    >
      {columns.map()}
    </InfiniteScroll>
    ...
```

File: src/components/ColumnPicker/ColumnPickerContent.tsx [hunk 2]
```diff
@@ -96,6 +96,7 @@ const ColumnPickerContent: React.FC<ColumnPickerContentProps> = props => {
   const classes = useStyles(props);
   const anchor = React.useRef<HTMLDivElement>();
   const scrollPosition = useElementScroll(anchor);
+  const scrollableTargetId = "columnPickerScrollableDiv";
```
Comment (jwm0) [inline src/components/ColumnPicker/ColumnPickerContent.tsx:99]: Define outside component

**Review by jwm0 — CHANGES_REQUESTED** [review]

File: src/components/ColumnPicker/ColumnPickerContent.tsx [hunk 3]
```diff
@@ -118,41 +119,25 @@ const ColumnPickerContent: React.FC<ColumnPickerContentProps> = props => {
...
-          useWindow={false}
-          threshold={100}
+          scrollThreshold="100px"
+          loader={
+            <div className={classes.loadMoreLoaderContainer}>
+              <CircularProgress size={16} />
+            </div>
+          }
+          scrollableTarget={scrollableTargetId}
           key="infinite-scroll"
```
Comment (jwm0) [inline src/components/ColumnPicker/ColumnPickerContent.tsx:138]: What's the `key` for?

File: src/orders/components/OrderProductAddDialog/OrderProductAddDialog.tsx [hunk 4]
```diff
@@ -267,18 +269,18 @@ const OrderProductAddDialog: React.FC<OrderProductAddDialogProps> = props => {
...
-      <DialogContent className={classes.content}>
+      <DialogContent className={classes.content} id={scrollableTargetId}>
         <InfiniteScroll
-          pageStart={0}
-          loadMore={onFetchMore}
+          dataLength={productChoicesWithValidVariants?.length}
+          next={onFetchMore}
           hasMore={hasMore}
-          useWindow={false}
+          scrollThreshold="100px"
```
Comment (jwm0) [inline src/orders/components/OrderProductAddDialog/OrderProductAddDialog.tsx:277]: We went from `10` to `100` in this particular component, is this intended?

File: src/components/AssignContainerDialog/AssignContainerDialog.tsx [hunk 5]
```diff
@@ -132,58 +127,57 @@ const AssignContainerDialog: React.FC<AssignContainerDialogProps> = props => {
...
+      </DialogContent>
+      <DialogContent
+        className={scrollableDialogClasses.scrollArea}
+        id={scrollableTargetId}
+      >
+        <InfiniteScroll
+          dataLength={containers?.length}
+          next={onFetchMore}
+          hasMore={hasMore}
+          scrollThreshold="100px"
```
Comment (jwm0) [inline src/components/AssignContainerDialog/AssignContainerDialog.tsx:139]: threshold went from `10` to `100`, intended?

File: src/orders/components/OrderProductAddDialog/OrderProductAddDialog.tsx (hunk 4 above)
Comment (orzechdev) [reply src/orders/components/OrderProductAddDialog/OrderProductAddDialog.tsx:277]: I've changed it as imo 10 is unreasonably small

File: src/components/AssignContainerDialog/AssignContainerDialog.tsx (hunk 5 above)
Comment (orzechdev) [reply src/components/AssignContainerDialog/AssignContainerDialog.tsx:139]: same here

File: src/components/ColumnPicker/ColumnPickerContent.tsx (hunk 3 above)
Comment (orzechdev) [reply src/components/ColumnPicker/ColumnPickerContent.tsx]: It's unneeded, removed.

**Review by jwm0 — APPROVED** [review]

**Review by eCzerniak — APPROVED** [review]

**Pull request merged**